src/test/test-lwm2m.c

```diff
@@ -562,94 +1322,522 @@ delete_cb(void *data,
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INIT(r, &sec_server[2], SERVER_OBJECT_BINDING,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_STRING, &binding);
+    ASSERT(r == 0);
+
+    printf("DBG: [Bootstrap Write /1]---> '%s'\n",
+        sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo));
+    r = sol_lwm2m_bootstrap_server_write_object(server, bs_cinfo, "/1",
+        servers, servers_len, servers_ids, sol_util_array_size(servers), write_servers_cb, NULL);
+    ASSERT(r == 0);
+
+    for (i = 0; i < sol_util_array_size(servers); i++)
+        for (j = 0; j < sol_util_array_size(nosec_server); j++)
+            sol_lwm2m_resource_clear(&servers[i][j]);
+}
+
+static void
+delete_all_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *bs_cinfo, const char *path,
+    enum sol_coap_response_code response_code)
+{
+    int r;
+    uint16_t i;
+    struct sol_lwm2m_resource sec_server[6];
+
+    ASSERT(response_code == SOL_COAP_RESPONSE_CODE_DELETED);
+    ASSERT(!strcmp("/", path));
+
+    // PSK-secured Server's Security Object
+    SOL_LWM2M_RESOURCE_SINGLE_INIT(r, &sec_server[0], SECURITY_OBJECT_SERVER_URI,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_STRING, &sec_server_dtls_addr);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INIT(r, &sec_server[1], SECURITY_OBJECT_IS_BOOTSTRAP,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_BOOL, false);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INT_INIT(r, &sec_server[2], SECURITY_OBJECT_SECURITY_MODE,
+        SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INIT(r, &sec_server[3], SECURITY_OBJECT_PUBLIC_KEY_OR_IDENTITY,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_STRING, &sec_server_psk_id);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INIT(r, &sec_server[4], SECURITY_OBJECT_SECRET_KEY,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_STRING, &sec_server_psk_key);
+    ASSERT(r == 0);
+    SOL_LWM2M_RESOURCE_SINGLE_INT_INIT(r, &sec_server[5], SECURITY_OBJECT_SERVER_ID, 102);
+    ASSERT(r == 0);
+
+    printf("DBG: [Bootstrap Write /0/0]---> '%s'\n",
+        sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo));
+    r = sol_lwm2m_bootstrap_server_write(server, bs_cinfo, "/0/0",
+        sec_server, sol_util_array_size(sec_server), write_sec_server_cb, NULL);
+    ASSERT(r == 0);
+
+    for (i = 0; i < sol_util_array_size(sec_server); i++)
+        sol_lwm2m_resource_clear(&sec_server[i]);
+}
+
+static void
+bootstrap_request_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *bs_cinfo)
+{
+    int r;
+
+    ASSERT(!strcmp("cli1", sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo)));
+
+    printf("DBG: [Bootstrap Delete /]---> '%s'\n",
+        sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo));
+    r = sol_lwm2m_bootstrap_server_delete_object_instance(server, bs_cinfo, "/",
```

Comment:
I guess we should add a test that a server will send a delete on '/' while NOT in bootstrap.